### PR TITLE
set mtime for fragmented videos (such as DASH)

### PR DESCRIPTION
**Pull request by dotysan**
### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

set mtime for fragmented videos (such as DASH)
 - it is set to the mtime of the last fragment uploaded

See issue #18384.

### CODE REVIEW

**Comment by dstftw** [comment]
This will set mtime for the final file *number-of-fragment* times. This is completely useless. mtime must be set one single time.

**Comment by dotysan** [comment]
> This will set mtime for the final file _number-of-fragment_ times.

Agreed. Although the mtime of the last fragment is what is effective after the download is complete.

> This is completely useless. mtime must be set one single time.

Is the call to os.utime() for each fragment really that expensive? I agree it would be more efficient to do it only once.

> Another point that appears immediately - why the last fragment's mtime and not the first's or 42th's?

Because that's the last time the file was modified. I.e. when the last fragment finally uploaded.

**Comment by dotysan** [comment]
OK @dstftw take another look at this PR. Indeed FileDownloader already calls os.utime() on every fragment, so there's no point in doing it twice on every fragment. I've stuffed it into a class variable and the syscall is just done once in _finish_frag_download() now.

**Comment by dotysan** [comment]
> mtime must be set one single time.

@dstftw BTW, it already calls os.utime() on *every* fragment here:
https://github.com/ytdl-org/youtube-dl/blob/2cd43a00d1eec54c40e3fdf9a288f4cb391a8323/youtube_dl/downloader/common.py#L222

 This is in FileDownloader.try_utime() called from HttpFD.download(). And *in this case of fragmented files* it's just as wasteful, since we only need the mtime of the last fragment.

If you are truly concerned with eliminating wasteful syscalls, we should fix this too.

File: youtube_dl/downloader/fragment.py [hunk 1]
```diff
@@ -116,6 +116,7 @@ def _append_fragment(self, ctx, frag_content):
         finally:
             if self.__do_ytdl_file(ctx):
                 self._write_ytdl_file(ctx)
+            self.mtime = os.stat(ctx['fragment_filename_sanitized']).st_mtime
```
Comment (dstftw) [inline youtube_dl/downloader/fragment.py:119]: Nothing conceptually changed: you still request `st_mtime` unnecessarily *n-1* times.

**Review by dstftw — CHANGES_REQUESTED** [review]

**Comment by dstftw** [comment]
> > mtime must be set one single time.
> 
> @dstftw BTW, it already calls os.utime() on _every_ fragment here:
> https://github.com/ytdl-org/youtube-dl/blob/2cd43a00d1eec54c40e3fdf9a288f4cb391a8323/youtube_dl/downloader/common.py#L222
> 
> This is in FileDownloader.try_utime() called from HttpFD.download(). And _in this case of fragmented files_ it's just as wasteful, since we only need the mtime of the last fragment.
> 
> If you are truly concerned with eliminating wasteful syscalls, we should fix this too.

Calling `os.utime` on each separate fragment is perfectly legal since fragments are not necessarily removed after the download.

**Pull request closed without merging**